### PR TITLE
Adding the ability to use proxy servers

### PR DESCRIPTION
**Pull request by romanpravda**
This PR adds the ability to add proxy servers (i.e. chproxy servers) and use them for queries by calling usingProxyServer function. This function chooses random proxy server from proxy servers in server provider.
Proxy servers may be added in server provider by addProxyServer function.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,6 +72,29 @@ By default client will use random server in given list of servers or in specifie
...
+
+$serverProvider = (new Tinderbox\Clickhouse\ServerProvider())->addServer($server)->addCluster($cluster);
+
+$client = (new Tinderbox\Clickhouse\Client($serverProvider));
+```
+
+To use server with tag, you should call ```usingServerFromGroup``` function before execute any query.
+
+```php
+$client->usingServerFromGroup('tag');
```
Comment (evsign) [inline README.md:95]: Not existed method.

Comment (romanpravda) [reply README.md]: Fixed that.

File: src/ServerProvider.php [hunk 2]
```diff
@@ -92,7 +123,20 @@ public function getServer(string $serverHostname): Server
...
+        if (!isset($this->serversByTags[$tag])) {
+            throw ServerProviderException::serverTagNotFound($tag);
+        }
+
+        if (!isset($this->serversByTags[$tag][$serverHostname])) {
+            throw ServerProviderException::serverHostnameNotFoundForTag($tag, $serverHostname);
+        }
+
+        return $this->serversByTags[$tag][$serverHostname];
+    }
```
[the remark this answers is not in the document]
Comment (romanpravda) [reply src/ServerProvider.php]: Removed this method.

File: src/Cluster.php [hunk 3]
```diff
@@ -116,12 +147,35 @@ public function getServerByHostname(string $hostname): Server
...
+        if (!isset($this->serversByTags[$tag])) {
+            throw ClusterException::tagNotFound($tag);
+        }
+
+        if (!isset($this->serversByTags[$tag][$hostname])) {
+            throw ClusterException::serverNotFoundByTag($tag, $hostname);
+        }
+
+        return $this->serversByTags[$tag][$hostname];
+    }
```
[the remark this answers is not in the document]
Comment (romanpravda) [reply src/Cluster.php]: Removed this method.

File: src/ServerProvider.php [hunk 4]
```diff
@@ -76,6 +98,15 @@ public function getRandomServerFromCluster(string $cluster): Server
         return $cluster->getServerByHostname($randomServerIndex);
     }
 
+    public function getRandomServerFromClusterByTag(string $cluster, string $tag): Server
+    {
+        $cluster = $this->getCluster($cluster);
+
+        $randomServerIndex = array_rand($cluster->getServersByTag($tag), 1);
+
+        return $cluster->getServerByHostname($randomServerIndex);
```
Comment (evsign) [inline src/ServerProvider.php:107]: Not seems that `$randomServerIndex` contains index. Technically thats maybe right, but semantically that variable store `hostname`.

Comment (romanpravda) [reply src/ServerProvider.php]: Changed variable name to ```$randomServerHostname```.

**Pull request merged**